network/p2p/dns/resolver.go

```diff
@@ -67,6 +65,26 @@ func WithTTL(ttl time.Duration) optFunc {
 	}
 }
 
+// WithResolverSize is an option function for setting the number of resolver workers and queue size.
+func WithResolverSize(numIPAddrLookupWorkers int, numTxtLookupWorkers int, ipAddrLookupQueueSize int, txtLookupQueueSize int) optFunc {
```

Comment:
Please have a separate method per each of these parameters, i.e., `WithIpWorkersCount(count int)`, `WithTxtWorkersCount(count int)`, `WithIpQueueSize(size int)` and `WithTxtQueueSize(size int)`. In general, we should not have high coupling in our option functions. Thanks.